test/ERC7432.spec.ts

```diff
@@ -0,0 +1,341 @@
+import hre, { ethers } from 'hardhat'
+import { Contract } from 'ethers'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { expect } from 'chai'
+import { ERC7432InterfaceId } from './contants'
+import nock from 'nock'
+import axios from 'axios'
+import { defaultAbiCoder, solidityKeccak256 } from 'ethers/lib/utils'
+import { NftMetadata, Role } from './types'
+
+const { HashZero, AddressZero } = ethers.constants
+const ONE_DAY = 60 * 60 * 24
+
+describe('ERC7432', () => {
+  let nftRoles: Contract
+  let nft: Contract
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  let deployer: SignerWithAddress
+  let roleCreator: SignerWithAddress
+  let userOne: SignerWithAddress
+  let userTwo: SignerWithAddress
+
+  const PROPERTY_MANAGER = solidityKeccak256(['string'], ['PROPERTY_MANAGER'])
+  const PROPERTY_TENANT = solidityKeccak256(['string'], ['PROPERTY_TENANT'])
+
+  const tokenId = 1
+  let nftMetadata: NftMetadata
+
+  before(async function () {
+    // prettier-ignore
+    [deployer, roleCreator, userOne, userTwo] = await ethers.getSigners()
+
+    const NftFactory = await ethers.getContractFactory('Nft')
+    nft = await NftFactory.deploy()
+    await nft.deployed()
+
+    const metadata: NftMetadata = {
+      name: 'Nft name',
+      description: 'Nft description',
+      roles: [
+        {
+          name: 'PROPERTY_MANAGER',
+          description: 'Property Manager',
+          supportsMultipleAssignments: false,
+          inputs: [
+            {
+              name: 'profitSplit',
+              type: 'tuple[]',
+              components: [
+                {
+                  name: 'eventId',
+                  type: 'uint256',
+                },
+                {
+                  name: 'split',
+                  type: 'uint256[]',
+                },
+              ],
+            },
+          ],
+        },
+        {
+          name: 'PROPERTY_TENANT',
+          description: 'Property Tenant',
+          supportsMultipleAssignments: true,
+          inputs: [
+            {
+              name: 'rentalCost',
+              type: 'uint256',
+            },
+          ],
+        },
+      ],
+    }
+
+    const scope = nock('https://example.com').get(`/${tokenId}`).reply(200, metadata)
+
+    const response = await axios.get(`https://example.com/${tokenId}`)
+    nftMetadata = response.data
```

Comment:
setup of mock server should happen on `before`, but the actual http call should be made during each test case.

You should also get the token uri from the view function of the ERC721 contract, not hard code it.